hitch.go

```diff
@@ -100,17 +100,14 @@ const paramsKey key = 1
 
 func wrap(handler http.Handler) httprouter.Handle {
 	return func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
-		httpcontext.Set(req, paramsKey, params)
+		ctx := context.WithValue(req.Context(), paramsKey, params)
+		*req = *req.WithContext(ctx)
 		handler.ServeHTTP(w, req)
```

Comment:
Why the pointer deference?

Reply:
I made this change because I was concerned about the effect on the upper middleware layer due to the change of pointer address of a request, but it's indeed better not to do dereference.

Thanks for the review. I fixed it.